.github/workflows/check-for-api-changes.yml

```diff
@@ -188,55 +188,18 @@ jobs:
       - name: Generate release notes
         if: steps.detect-changes.outputs.changes_detected == 'true'
         id: release-notes
-        uses: vercel/ai-action@v2
-        with:
-          model: "openai/gpt5"
-          api-key: ${{ secrets.AI_GATEWAY_API_KEY }}
-          prompt: |
-            Analyze the following OpenAPI specification diff and calculate these three things
-
-            1. Summary of changes as conventional commit messages (max. 100 characters)
-            2. A detailed description of the changes, in markdown format
-            3. The type of version change: "breaking", "feature", or "fix"
-
-            For the markdown description, create three sections, each with its own level 3 header:
-
-            1. Breaking changes (removed endpoints, parameters, or changed behavior)
-            2. New features (new endpoints, parameters, or options)
-            3. Fixes (documentation updates, typo corrections)
-
-            For each section, group changes by their routes. Derive the route from the file path in the diff. For example, changes in `cache/cohere/routes/v2/chat/post.json` belong to the `POST /v2/chat` route.
-
-            If a section has no content, omit it from the release notes.
-
-            Do not list items in new features that are listed in breaking changes.
-            Do not list items in fixes that are listed in breaking changes or new features.
-
-            Diff:
-            ${{ steps.diff.outputs.text }}
-          schema: |
-            {
-              "$schema": "http://json-schema.org/draft-07/schema#",
-              "title": "Change Description",
-              "type": "object",
-              "properties": {
-                "summary": {
-                  "type": "string",
-                  "description": "Summary of changes as conventional commit messages (max. 100 characters)."
-                },
-                "description": {
-                  "type": "string",
-                  "description": "Description of the change, in markdown format."
-                },
-                "version": {
-                  "type": "string",
-                  "enum": ["breaking", "feature", "fix"],
-                  "description": "The type of version change."
-                }
-              },
-              "required": ["summary", "description", "version"],
-              "additionalProperties": false
-            }
+        env:
+          OPENAI_API_KEY: ${{ secrets.AI_GATEWAY_API_KEY }}
+        run: |
+          # Generate release notes using custom script
+          node scripts/generate-release-notes.js "${{ steps.diff.outputs.text }}" "${{ matrix.source.name }}"
+          
+          # Read the output and set GitHub Actions outputs
+          RELEASE_OUTPUT=$(cat release-notes-output.json)
+          
+          echo "json<<EOF" >> $GITHUB_OUTPUT
+          echo "$RELEASE_OUTPUT" >> $GITHUB_OUTPUT
+          echo "EOF" >> $GITHUB_OUTPUT
```

Comment:
I think we should have the script write directly to output variables using the `@actions/core` module

https://github.com/actions/toolkit/tree/main/packages/core#readme